api/envoy/extensions/filters/http/thrift_to_metadata/v3/thrift_to_metadata.proto

```diff
@@ -83,7 +81,9 @@ message FieldSelector {
 
 // [#next-free-field: 6]
 message Rule {
-  // The field to match on. If set, takes precedence over field_selector.
+  // The field to match on.
+  // :ref:`field_selector<envoy_v3_api_field_extensions.filters.http.thrift_to_metadata.v3.Rule.field_selector>`
+  // takes precedence if both are set.
```

Comment:
seems like this is a behavior change, can you add a release note about it ? can you help me understand if this is an incompatible behavior change?

Reply:
People can use WIP fields: we have validators here https://github.com/envoyproxy/envoy/blob/0440ea1a38bbb8932732428010708d29f39d43b7/source/common/protobuf/message_validator_impl.cc#L38 that warn about them using it and the field themselve can arbitrarily change https://github.com/envoyproxy/envoy/blob/main/api/API_VERSIONING.md#backwards-compatibility

but this is a behavior change in that it seems the `field` took precedence over over `field_selector` but now that has flipped.